internal/mtail/multiple_levels_directory_integration_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+// This file is available under the Apache license.
+// +build integration
+
+package mtail_test
+
+import (
+	"os"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/mtail/internal/mtail"
+	"github.com/google/mtail/internal/testutil"
+)
+
+func TestPollLogPathPatterns(t *testing.T) {
+	tmpDir, rmTmpDir := testutil.TestTempDir(t)
+	defer rmTmpDir()
+
+	logDir := path.Join(tmpDir, "logs")
+	progDir := path.Join(tmpDir, "progs")
+	testutil.FatalIfErr(t, os.Mkdir(logDir, 0700))
+	testutil.FatalIfErr(t, os.Mkdir(progDir, 0700))
+	defer testutil.TestChdir(t, logDir)()
+
+	m, stopM := mtail.TestStartServer(t, 10*time.Millisecond, false, mtail.ProgramPath(progDir), mtail.LogPathPatterns(logDir+"/files/*/log/*log"))
+	defer stopM()
+
+	startLogCount := mtail.TestGetMetric(t, m.Addr(), "log_count")
+	startLineCount := mtail.TestGetMetric(t, m.Addr(), "lines_total")
+
+	logFile := path.Join(logDir, "files", "a", "log", "a.log")
+	testutil.FatalIfErr(t, os.MkdirAll(path.Dir(logFile), 0700))
+	f := testutil.TestOpenFile(t, logFile)
+	n, err := f.WriteString("line 1\n")
+	testutil.FatalIfErr(t, err)
+	glog.Infof("Wrote %d bytes", n)
+	time.Sleep(2 * time.Second)
```

Comment:
The test failed, I suspect because of a race condition waiting for mtail to read them all.

Look for 'DoOrTimeout' which runs a check every interval until a timeout, every 100ms and timeout at 10s is what I usually do I think.  Then you can put the following lines in the check function, and they'll get run until they pass or timeout.